src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1771,6 +1789,44 @@ private void RegisterHomeWorkspace(HomeWorkspaceModel newWorkspace)
             };
         }
 
+        /// <summary>
+        /// Collects top level assemblies referenced by nodes in the current workspace
+        /// </summary>
+        /// <returns></returns>
+        private Dictionary<Guid, AssemblyName> OnCollectingAssembliesReferencedByNodes()
+        {
+            var assemblyNames = new Dictionary<Guid, AssemblyName>();
+            foreach(var node in CurrentWorkspace.Nodes)
+            {
+                // Get zerotouch assemblies
+                if (node.GetType() == typeof(DSFunction))
+                {
+                    // Split CreationName to get qualified name
+                    var qualifiedName = node.CreationName.Split('@')[0];
```

Comment:
Both the `CreationName` and `FunctionSignature` look like this: `HowickMaker.hMember.ByLineVector@Autodesk.DesignScript.Geometry.Line,Autodesk.DesignScript.Geometry.Vector,string`. But you need just the part on the left of the "@" (`HowickMaker.hMember.ByLineVector`) to look up the function descriptors. Is there an easier way to do this?

Reply:
Like Mike says I have a feeling this can be very slow. Further string operations make things slower so if there is a way to avoid string manipulation we should aim for that.

Reply:
@aparajit-pratap @mjkkirschner I was able to simplify this significantly: https://github.com/DynamoDS/Dynamo/pull/9723/commits/d6d35299fe5c1293f63c9781fe3580fad4808dae